File: Previous Rewrite/CombatTracker.DomainOld/Reference/Attacks/Charts/CriticalType.cs
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;
using CombatTracker.Domain.Reference.Base;

namespace CombatTracker.Domain.Reference.Attacks.Charts
{
    [Table("Attack_CriticalTypes")]
    public partial class CriticalType : ReferenceEntry
    {
        public CriticalType()
        {
            //this.Attacks = new HashSet<Attack>();
            //this.AttacksAddCrits = new HashSet<AttacksAddCrit>();
        }
    
        public override string ToString()
        {
            return Name;
        }

        //public static CriticalType LoadByName(RmssDbContext db, string name)
        //{
        //    return (from sc in db.CriticalTypes where sc.Name == name select sc).FirstOrDefault();
        //}
    }
    
}

File: CombatTracker-Application/CombatTrackerData/CreatureCharts/CriticalType.vb
Partial Class CriticalType
    Inherits IEntity(Of CriticalType)

    Public Overrides Function ToString() As String
        Return Name
    End Function

    Public Shared Function LoadByName(ByVal DB As RMSSDataDataContext, ByVal Name As String) As CriticalType
        Return (From CT In DB.CriticalTypes Where CT.Name = Name Select CT).FirstOrDefault
    End Function
End Class
